src/main/java/org/generationcp/middleware/dao/AttributeDAO.java

```diff
@@ -115,6 +135,58 @@
 		return returnList;
 	}
 
+	public long countAttributeValueDtos(final AttributeValueSearchRequestDto attributeValueSearchRequestDto, final String programUUID) {
+		final StringBuilder sql = new StringBuilder(" SELECT COUNT(DISTINCT cv.cvterm_id) ");
```

Comment:
Shouldn't the count be in on attribute values, not the attribute variable (cvterm_id)?